include/mbgl/util/interpolate.hpp

```diff
@@ -15,31 +15,31 @@
 namespace mbgl {
 namespace util {
 
-float interpolationFactor(float base, Range<float> range, float z);
+float interpolationFactor(float base, Range<float> range, float z) noexcept;
 
 template <class T, class Enabled = void>
 struct Interpolator;
 
 template <typename T>
-T interpolate(const T& a, const T& b, const double t) {
+T interpolate(const T& a, const T& b, const double t) noexcept {
     return Interpolator<T>()(a, b, t);
 }
 
 template <typename T>
-T interpolate(const T& a, const T& b, const float t) {
+T interpolate(const T& a, const T& b, const float t) noexcept {
     return Interpolator<T>()(a, b, t);
 }
 
 template <class T, class Enabled>
 struct Interpolator {
-    T operator()(const T& a, const T& b, const double t) const { return a * (1.0 - t) + b * t; }
+    T operator()(const T& a, const T& b, const double t) const noexcept { return a * (1.0 - t) + b * t; }
 };
 
 template <>
 struct Interpolator<float> {
-    float operator()(const float& a, const float& b, const float t) const { return a * (1.0f - t) + b * t; }
+    float operator()(const float& a, const float& b, const float t) const noexcept { return a * (1.0f - t) + b * t; }
```

Comment:
```suggestion
    float operator()(const float& a, const float& b, const float t) const { return a * (1.0f - t) + b * t; }
```

may call throwing copy constructor